darwin/dataset/utils.py

```diff
@@ -498,49 +502,61 @@ def get_annotations(
     )
 
 
-def _validate_inputs(partition, split_type, annotation_type):
+def _validate_inputs(
+    partition: Union[str, None], split_type: str, annotation_type: str
+) -> None:
     """
     Validates the input parameters for partition, split_type, and annotation_type.
 
     Args:
-        partition (str, None): Dataset partition. Should be 'train', 'val', 'test' or None.
-        split_type (str, None): Type of dataset split. Can be 'random', 'stratified' or None.
+        partition (str, None): Dataset partition. Should be 'train', 'val', 'test', or None.
+        split_type (str, None): Type of dataset split. Can be 'random', 'stratified', or None.
         annotation_type (str): Type of annotations. Can be 'tag', 'polygon', or 'bounding_box'.
 
     Raises:
         ValueError: If the input parameters do not match the expected values.
     """
     if partition not in ["train", "val", "test", None]:
-        raise ValueError("partition should be either 'train', 'val', 'test', or None")
-    if split_type not in ["random", "stratified", None]:
-        raise ValueError("split_type should be either 'random', 'stratified', or None")
+        raise ValueError("partition should be either 'train', 'val', 'test', or 'None'")
+    if split_type not in ["random", "stratified"]:
+        raise ValueError("split_type should be either 'random', or 'stratified'")
     if annotation_type not in ["tag", "polygon", "bounding_box"]:
         raise ValueError(
             "annotation_type should be either 'tag', 'bounding_box', or 'polygon'"
         )
 
 
-def _get_stems_from_split(release_path, split, split_type, annotation_type, partition):
+def _get_annotation_filepaths_from_split(
+    release_path: Path,
+    annotation_type: str,
+    partition: str,
+    split_type: str,
+    split: Optional[str] = "default",
+) -> Generator[str, None, None]:
     """
-    Determines the file stems based on the dataset split and other parameters.
+    Determines the filepaths based on the dataset split and other parameters.
```

Comment:
This function used to work with file path stems. However, to support datasets with identically named items in different folders, splits have to record the full file path to each annotation file